driver/driver.py

```diff
@@ -0,0 +1,181 @@
+import subprocess
+import wxflow
+from wxflow import Logger, Jinja
+from wxflow.configuration import cast_as_dtype
+
+def generate_eva_config(template_path: str, output_path: str, context: dict):
+    """
+    Render a Jinja template to create an EVA configuration file.
+
+    Args:
+        template_path (str): Path to the Jinja template file.
+        output_path (str): Path where the rendered config file will be saved.
+        context (dict): Dictionary containing the data used for rendering the template.
+
+    Returns:
+        str: Path to the rendered EVA configuration file.
+    """
+    jinja_render = Jinja(template_path_or_string=template_path, data=context)
+    jinja_render.save(output_file=output_path)
+    return output_path
+
+def copy_plots_to_public(runtime_dir: Path, logger, public_root: Path = Path("/public") / "plots"):
+    """
+    Copies all plot PNGs from runtime_dir/plots/** into the public/plots/ directory,
+    preserving all subdirectory structure.
+
+    Args:
+        runtime_dir (Path): Base runtime directory (which contains 'plots/' subfolder).
+        logger: Logger instance.
+        public_root (Path): Target base directory (default: 'public/plots/').
+    """
+    plots_dir = runtime_dir / "plots"
+    if not plots_dir.exists():
+        logger.warning(f"No 'plots/' directory found in runtime_dir: {runtime_dir}")
+        return
+
+    for plot_file in plots_dir.rglob("*.png"):
+        rel_path = plot_file.relative_to(plots_dir)
+        target_path = public_root / rel_path
+
+        target_path.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(plot_file, target_path)
+        logger.info(f"Copied {plot_file} → {target_path}")
+        
+def run_monitoring_job(args):
+    """
+    Main driver function for observation monitoring workflow.
+
+    This function:
+    - Creates a timestamped runtime directory
+    - Copies NetCDF files matching the observation type to that directory
+    - Generates a Jinja2-based EVA configuration file
+    - Runs the EVA command using the config
+    - (Optional) Copies the results to an output directory
+
+    Args:
+        args (tuple): A tuple containing:
+            - monitor_dict (dict): A dictionary with monitoring configuration details.
+            - timestamp (str): A timestamp string in the format "%Y%m%d_%H%M%S".
+    """
+    monitor_dict, timestamp = args
+
+    satellite = monitor_dict["satellite"]
+    sensor = monitor_dict["sensor"]
+    ob_type = f"{sensor}_{satellite}"
+
+    logger = Logger(f"Obs Monitor - {ob_type}")
+    logger.info("Starting Observation Monitoring")
+
+    experiment_dir = Path(os.getenv("EXPDIR"))
+    if not experiment_dir.exists():
+        raise FileNotFoundError(f"Experiment directory not found: {experiment_dir}")
+
+    runtime_root = Path(os.getenv("RUNTIME_DIR"))
+    template_path = monitor_dict["template_path"]
+
+    runtime_dir = runtime_root / f"runtime_{ob_type}_{timestamp}"
```

Comment:
os.path.join

Reply:
I would argue that the current implementation using a Path() object with the forward slash is more modern and preferred and in my opinion more readable by a user than `os.path.join()`. Performance wise is pretty much negligible difference as well.